### ISSUE TITLE
Expand Row unable to exclude column with dataformat 

### ISSUE THREAD

**Issue by steng2000** [AllenFang/react-bootstrap-table#1552]
Love react-bootstrap-table   

I am able to get expand row on column to work.   And I am able to exclude column e.g. 
```<TableHeaderColumn expandable={false} dataField='foo' ></TableHeaderColumn>```

However, when I added dataFormat to the TableHeaderColumn, expandable={false} does not work. 
```<TableHeaderColumn expandable={false} dataField='foo' **dataFormat={this.sizeFormatter}**></TableHeaderColumn>```

Are there specific practice I have to perform in order to have column with dataFormat excluded from row expansion. 

<pre>        const selectRow = {
            mode: 'checkbox',
            clickToExpand: true,
            clickToSelect: false,
            selected: this.state.selected,
            onSelect: this.onRowSelect,
            onSelectAll: this.onSelectAll
        }

        //options for the React bootstrap modal
        const options = {
            expandRowBgColor: 'rgb(242, 255, 163)',
            expandBy: 'column'
        };
</pre>

**Comment by AllenFang** [comment]
@steng2000 sorry for lately reply, I'll have a look on it, seems like a bug or something I can improve, thanks

**Comment by AllenFang** [comment]
it's a bug, I will fix it soon, thank you

**Comment by AllenFang** [comment]
Hi @steng2000, fixed on `v4.0.2`, thanks

**Comment by darinone** [comment]
@AllenFang 
Hi, i'm using 4.3.1 version of react-bootstrap-table. 
and when i click on an <a> row expanded despite on  expandable={false}. Can you help?

```
<TableHeaderColumn
                  editable={{ validator: ModifyRowValidators.emailValidator }}
                  id="email"
                  row="1"
                  dataField="email"
                  dataFormat={colFormatter}
                  dataSort
                  expandable={false}
                  width="250"
                >
```
```
    const colFormatter = (cell) => {
      return (
        <a href={`mailto:${cell}`}>{cell}</a>
      );
    };
```
